pkg/deploy/internal/setting/validation.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package setting
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log/field"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/entities"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/parameter"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/parameter/value"
+)
+
+type Validator struct{}
+
+var deprecatedSchemas = map[string]string{
+	"builtin:span-attribute":       "this setting was replaced by 'builtin:attribute-allow-list' and 'builtin:attribute-masking'",
+	"builtin:span-event-attribute": "this setting was replaced by 'builtin:attribute-allow-list' and 'builtin:attribute-masking'",
+}
+
+// Validate checks for each settings type whether it is using a deprecated schema and if a value type scope is a non-empty string.
+func (v *Validator) Validate(c config.Config) error {
+
+	s, ok := c.Type.(config.SettingsType)
+	if !ok {
+		return nil
+	}
+
+	if msg, deprecated := deprecatedSchemas[s.SchemaId]; deprecated {
+		log.WithFields(field.Coordinate(c.Coordinate), field.Environment(c.Environment, c.Group)).Warn("Schema %q is deprecated - please update your configurations: %s", s.SchemaId, msg)
+	}
+
+	return validateScopeType(c)
+}
+
+func validateScopeType(c config.Config) error {
```

Comment:
I am not sure if we should split the scope validation like this. I understand the intention, but as you are already writing in the commit message, it's very limited. 

I would rather remove this and fail during a settings deploy if the value parameter can't be resolved correctly. This change would remove the duplicate effort and split it into two parts vs having the error later but only one scope validation step

Reply:
As is we do not report this configuration error on dry-run, but only when you deploy. That is a horrible user experience. IMO the limited check is the absolute minimum we should do for a dry-run - better would be to fully validate the actual content and follow references, but that is more effort than I wanted to include in this unrelated change

Reply:
I was wrong, it's checked in dry-runs as well already